### ISSUE TITLE
Add support for submitting application jar by specifying application jar as URL

### ISSUE THREAD

**Issue by ColonelLanda** [big-data-europe/docker-spark#50]
At the moment the submit docker image requires the application jar to be available to the image. I would like to use the submit image for submitting application to a remote cluster where the application jar is in S3. Using the following command:
```
docker run --rm -it \
    -e ENABLE_INIT_DAEMON=false \
    -e SPARK_MASTER_NAME=$SPARK_MASTER \
    -e SPARK_MASTER_PORT=7077 \
    -e "SPARK_APPLICATION_JAR_LOCATION=http://s3-bucket/my.jar" \
    -e SPARK_APPLICATION_MAIN_CLASS=com.armis.aggregations.Main \
    spark-submit:spark-2.3.2-hadoop2.7
```

**Comment by atomobianco** [comment]
+1 for this one

I guess the first barrier to remove is the `if [ -f "${SPARK_APPLICATION_JAR_LOCATION}" ];` which forces the jar file to be a local file

**Comment by Chrescht1337** [comment]
I'm trying to pass a jar with the --packages argument, I guess that's the same issue? putting "--packages org.apache.spark:JARthing" in SPARK_APPLICATION_ARGS doesn't help

**Comment by Chrescht1337** [comment]
I discovered the env variable SPARK_SUBMIT_ARGS and added my --packages argument in there, which resulted in downloading the jar, but then I got : java.lang.ClassNotFoundException: my.main.Application
did you have similar experience? I didnt see any usage of SPARK_SUBMIT_ARGS anywhere else so I considered it safe to manipulate

**Comment by YevheniiStepaniuk** [comment]
@Chreschtf How you pass `SPARK_SUBMIT_ARGS ` into Template?

**Comment by GezimSejdiu** [comment]
Fixed via a248a2f781c8de82023745cffc60325eccda8ae9
